rust/kernel/sysctl.rs

```diff
@@ -150,7 +145,7 @@ impl<T: SysctlStorage> Sysctl<T> {
             data: &*storage as *const T as *mut core::ffi::c_void,
             proc_handler: Some(proc_handler::<T>),
 
-            maxlen: 0,
+            maxlen: mem::size_of::<T>() as i32,
```

Comment:
I think this needs to be configurable by the actual storage. For example an u32 is 4 bytes in size, but would write up to 10 characters if written in decimal.